### ISSUE TITLE
Rpcs are resent twice if the region is already NSRED and it gets cleared afterwards

### ISSUE THREAD

**Issue by dvdreddy** [OpenTSDB/asynchbase#81]
Client Version: 1.5.0
Sever Version :  -

<h2> BackGround : </h2>

<h5> Scenario 1 </h5>

When rpc is called sendToRegion and its region is KnownToBeNSREd then handleNSRE is called but at the same time RetryRpc callback is attached which will resend the rpc even after the rpc succeeded when the NSRE is finished. This will be mainly problematic with the atomicIncrement requests

<h5> Scenario 2 </h5>

when probe rpc expires with the number of tries (it will happen in around approx 10 sec) its call back chain will start executing, in which case all the rpcs in that are in the NSRE list  will be called sendRpcToRegion assuming that NSRE is cleared. But in this path if the first few rpc's response returns with region still being NSREd before the other rpc's client.sendRpc is triggered (this is quite possible if above 1000 rpcs are waiting on NSRE because these rpcs may bewaiting on meta region lookup and before this clears up, the first few rpcs may have returned NSRE Exception from the region server) all the remaining rpcs will go through knownTo be NSREd codepath in which retryRpc callback will be added to these rpc's deferred in which case all of them will be resent again after the clearing of NSRE. This can be disastrous in the case of probe rpc getting expired a few number of times. (SideNote : This happend once in our production cluster where the probe expired 5 times and each of waiting rpcs we tracked were sent to the server 6 times)

<h2> How to reproduce </h2>

sent the pull request #80  which two tests which reproduces the above explained scenario. 

<h2> fix </h2> 

the only fix I can think of is the removal of retryRpc() callback during the knownToBeNSREd code path
